com.tibia.Tibia.metainfo.xml

```diff
@@ -0,0 +1,44 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<component type="desktop-application">
+  <id>com.tibia.Tibia</id>
+
+  <name>Tibia</name>
+  <summary>Become a part of Tibia, the MMORPG classic that has been expanded for more than 25 years</summary>
+
+  <metadata_license>CC0-1.0</metadata_license>
+  <project_license>LicenseRef-proprietary</project_license>
```

Comment:
Does this allow redistribution ? If not you have to use extra-data https://docs.flatpak.org/en/latest/module-sources.html#extra-data

Reply:
Unclear, it's not mentioned anywhere on the website that redistribution is not allowed, and many websites redistributed files over the years, especially before the launcher.

If it is valuable, I can try to contact the company again about redistribution policy.

Reply:
If they don't explicitly say it's allowed, it means it's not and you have to use extra-data.

If you get permission from them then it's fine. You can contact them.

Reply:
I will try to get their permission then! Will return in a few days since they don't reply on weekends :wink:

Reply:
If you're not able to get permission for distribution from them, I'm pretty sure it'll be impossible to use the official tarball as a source, because its hash will always be changing.

However, given that the game seems to have a built-in auto-update feature, you could probably archive the link of the tarball to [web.archive.org](https://web.archive.org) and use that as a source instead...

Reply:
That won't do... it's still the official tarball but with a different non official link.

Reply:
I understand, but why does Flathub take issue with that? For all intents and purposes, it's the official program, just from a different (and trustworthy) mirror...

Reply:
Because you will sill be distributing their binaries/sources now from a different link. changing the hosting place does not mean their terms and conditions over it disappear.

Reply:
I think I wasn't clear in my previous post. I only suggested archiving the tarball if @ranisalt didn't get permission to redistribute from them, i.e. his only option is to use `extra-data`.

If he gets permission to distribute, this checksum issue won't be a major problem unless Flathub requires reproducible builds.

Reply:
If they don't have permission to distribute the original link can be used with extra-data. Why does it need archiving ?

We will allow automerge in such cases so that the Flatpak remains installable.

Reply:
Ok I missed the reply about x-checker below. I see what you meant now.

Reply:
>Why does it need archiving ?

Because unless Flathub is able to keep the tarball checksum in-sync with upstream (down to the second), some unlucky Flathub users will inevitably have issue while installing the game simply because the hash of the tarball doesn't match the one expected in `extra-data`...

Reply:
Considering f-e-d-c is currently unable to handle the requirements to fetch the file (see below), I'm inclined to use an archive of the file. That would avoid the checksum sync issue, and it will auto-update anyway, one does not need the most up to date launcher at all times.

Is there any facility in place to trigger Internet Archive to redownload a file every once in a while?

Reply:
>one does not need the most up to date launcher at all times.

Right. Though I think you'd need a wrapper launcher script that would copy all launcher files from `/app/bin` to a place where they can be overwritten (`$XDG_DATA_HOME` is a common choice).

Similarly to [what JDownloader does](https://github.com/flathub/org.jdownloader.JDownloader/blob/master/jd-wrapper.sh).

>Is there any facility in place to trigger Internet Archive to redownload a file every once in a while?

I'm not aware of such tool. If I were to maintain this package, I'd use [nvchecker](https://github.com/lilydjwg/nvchecker) to periodically monitor `https://static.tibia.com/launcher/launcher-linux-current/package.json` for changes on the `version` key. Once that key changed, I'd just manually re-archive their `.tar.gz` on [web.archive.org](https://web.archive.org/).